velox/common/time/Timer.h

```diff
@@ -37,6 +40,28 @@ class MicrosecondTimer {
   uint64_t* timer_;
 };
 
+class ClockTimer {
```

Comment:
This class is very similar to MicrosecondTimer, but uses folly::hardware_timestamp instead of std::chrono::steady_clock::now() . Would you add comments to both of these classes to explain the difference and provide guidance of when to use which one. Also, I wonder if you can think of a more descriptive names for these classes. Finally, ClockTimer allows to update atomic variable, but MicrosecondTimer does not. Any reason not to allow MicrosecondTimer update atomic variable? If yes, would you add comments to explain why.